ethcore/src/block.rs

```diff
@@ -278,10 +282,55 @@ impl<'x> OpenBlock<'x> {
 		let gas_floor_target = cmp::max(gas_range_target.0, engine.params().min_gas_limit);
 		let gas_ceil_target = cmp::max(gas_range_target.1, gas_floor_target);
 		engine.populate_from_parent(&mut r.block.header, parent, gas_floor_target, gas_ceil_target);
+		Self::push_last_hash(&mut r.block, last_hashes, engine, &parent.hash())?;
```

Comment:
a utility for engines to call contract functions before and after the transactions within a block as a part of the state transitions seems like it would be useful generally